sympy/solvers/recurr.py

```diff
@@ -614,11 +614,14 @@ def rsolve_hyper(coeffs, f, n, **hints):
             if coeff is not S.Zero:
                 poly += coeff * Z**i
 
+        constants = []
+
         for z in roots(poly, Z).keys():
             if z.is_zero:
                 continue
 
-            (C, s) = rsolve_poly([polys[i].as_expr()*z**i for i in range(r + 1)], 0, n, symbols=True)
+            (C, s) = rsolve_poly([polys[i].as_expr()*z**i for i in range(r + 1)], 0, n, len(constants), symbols=True)
+            constants.extend(s)
```

Comment:
Are the renumbered constants are always independent of the existing constants? If not, it would be possible to get incorrect solutions by substituting values for the constants that don't satisfy the correct relationships.

Reply:
From what I understand, we're just combining multiple solutions, so similar to ODE's, we do the `c0*f0(x) + c1*f1(x)` and so on. The problem here was the naming of constants always started with 0 for each individual solution which led to total no. of constants being lesser than the order. So, I don't think there would be anything wrong, but I'll test it with some more higher order equations and see if it fails anywhere